README.adoc

```diff
@@ -70,156 +60,77 @@ This guide is a work in progress - existing guidelines are constantly being impr
-
-Originally, `git ls-files` inside the default gemspec template was designed so that users
-publishing their first gem wouldn't unintentionally publish artifacts to it.
-Recent versions of bundler won't let you release if you have uncommitted files in your
-working directory, so that risk is lower.
-
-==== Examples [[gemspec-git-examples]]
+Avoid using `git ls-files` to produce lists of files. Downstreams (OS packagers) often
+need to build your package in an environment that does not have git (on purpose).
+Instead, use some pure Ruby alternatives, like `Dir` or `Dir.glob`.
 
 [source,ruby]
 ----
 # bad
 Gem::Specification.new do |spec|
-  spec.files         = `git ls-files`.split('\n')
-  spec.test_files    = `git ls-files -- spec/*`.split('\n')
+  spec.files         = `git ls-files`.split("\n")
+  spec.test_files    = `git ls-files -- spec`.split("\n")
 end
 
 # good
 Gem::Specification.new do |spec|
-  spec.files         = Dir['lib/**/*', 'LICENSE', 'README.md']
-  spec.test_files    = Dir['spec/**/*']
+  spec.files         = Dir["lib/**/*", "LICENSE", "README.md"]
+  spec.test_files    = Dir["spec/**/*"]
 end
 
 # bad
 Gem::Specification.new do |spec|
-  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
-    `git ls-files -z`.split('\\x0').reject { |f| f.match(%r{^(test|spec|features)/}) }
+  spec.files = Dir.chdir(File.expand_path(__dir__)) do
+    `git ls-files -z`.split("\\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
   end
 end
 
 # good
 Gem::Specification.new do |spec|
-  spec.files         = Rake::FileList['**/*'].exclude(*File.read('.gitignore').split)
+  spec.files         = Rake::FileList["**/*"].exclude(*File.read(".gitignore").split)
 end
 
 # bad
 Gem::Specification.new do |spec|
-  spec.files         = `git ls-files`.split('\n')
-  spec.test_files    = `git ls-files -- test/{functional,unit}/*`.split('\n')
-  spec.executables   = `git ls-files -- bin/*`.split('\n').map { |f| File.basename(f) }
+  spec.files         = `git ls-files -- lib/`.split("\n")
+  spec.test_files    = `git ls-files -- test/{functional,unit}/*`.split("\n")
+  spec.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
 end
 
 # good
 Gem::Specification.new do |spec|
-  spec.files         = Dir.glob('lib/**/*')
-  spec.test_files    = Dir.glob('test/{functional,test}/*')
-  spec.executables   = Dir.glob('bin/*').map { |f| File.basename(f) }
+  spec.files         = Dir.glob("lib/**/*")
+  spec.test_files    = Dir.glob("test/{functional,test}/*")
+  spec.executables   = Dir.glob("bin/*").map{ |f| File.basename(f) }
 end
 ----
 
+=== Using require_relative from test code into lib/ [[using-require-relative-from-test-to-lib]]
 
-=== RequireRelativeHardcodingLib: Using require_relative from test code into lib/ [[require-relative-hardcoding-lib]]
-
-Avoid using `require_relative` with relative path to lib. Use `require` instead.
-
-==== Rationale [[require-relative-hardcoding-lib-rationale]]
```

Comment:
You can keep rationales like this one IMO, but it's up to you.

Reply:
Your advice is really valuable, thanks for that!
I've re-added them now! 😄